protobufs/livekit_models.proto

```diff
@@ -302,6 +305,26 @@ message ChatMessage {
   bool generated = 6; // true if the chat message has been generated by an agent from a participant's audio transcription
 }
 
+message RpcRequest {
+  string id = 1;
+  string method = 2;
+  string payload = 3;
+  uint32 response_timeout_ms = 4;
+  uint32 version = 5;
+}
+
+message RpcAck {
+  string request_id = 1;
+}
+
+message RpcResponse {
+  string request_id = 1;
+  oneof value {
+    string payload = 2;
+    string error = 3;
```

Comment:
in a previous iteration you had `Error` as consisting of both an error_message and an error_code (or similar). What's the incentive of dropping the error_code? I thought that was a neat addition.

Reply:
The ergonomics were messy in TypeScript, although maybe I was just doing it wrong. It's awkward to `reject` a promise with anything but an `Error` object, and populating code and data from that reliably was also awkward. Maybe what I'm doing now where if you throw an error the `message` is automatically brought along is also problematic. We don't really want to bring a whole random error object along the wire, so probably the framework should be catching and translating most errors. I'll think about it some more.

I also found that a developer would have to lookup our codes to know what a given error means, whereas using string identifiers makes it self-evident, whether in the debugger or in logs. (of course we'd still recommend they use our constants to actually write error handling code).

Reply:
on the JS side we are extending the native `Error` with a `LiveKitError` class that includes a error code, maybe a similar pattern could work here

Reply:
not sure if you're suggesting the error is _only_ a string message, a string status or both.

Reply:
Yeah right now it's a single string. Another complication is that this has to work across every language: an error thrown in the swift SDK needs to make it across and come out the other side in python, etc. I haven't done a detailed yet but I'm more confident in the portability of a single string rather than a code + message.

I updated the prototype to make this cleaner and also to catch unexpected errors thrown in the handler, which were previously getting sent across the wire and that seemed wrong: https://github.com/livekit/node-sdks/pull/276/files#diff-6b1f6b4d4de8fecd4d8dbf20eb3b46edd692b6fc4681b694172ad43a02862fcfR59